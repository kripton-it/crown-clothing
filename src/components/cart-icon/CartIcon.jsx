import React from "react";

import { CartIconContainer, ShoppingIcon, ItemsCount } from "./CartIconStyles";

const CartIcon = ({ toggleCartHidden, itemsCount, location }) => {
  return (
    <CartIconContainer
      onClick={() =>
        location.pathname !== "/checkout" && itemsCount && toggleCartHidden()
      }
    >
      <ShoppingIcon />
      <ItemsCount>{itemsCount}</ItemsCount>
    </CartIconContainer>
  );
};

export default CartIcon;
