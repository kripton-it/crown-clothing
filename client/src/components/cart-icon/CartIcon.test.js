import React from "react";
import { shallow } from "enzyme";

import CartIcon from "./CartIcon";

describe("CartIcon component", () => {
  let wrapper;

  it("should render", () => {
    const mockProps = {
      location: {
        pathname: "/"
      },
      toggleCartHidden: jest.fn(),
      itemsCount: 0
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the itemCount as the text', () => {
    const mockProps = {
      location: {
        pathname: "/"
      },
      toggleCartHidden: jest.fn(),
      itemsCount: 5
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
    const itemCount = parseInt(wrapper.find('ItemsCount').text());
    expect(itemCount).toBe(5);
  });

  it('should call toggleCartHidden when CartIcon is clicked, if itemsCount > 0 and pathname !== "/checkout"', () => {
    const mockToggleCartHidden = jest.fn();
    const mockProps = {
      location: {
        pathname: "/"
      },
      toggleCartHidden: mockToggleCartHidden,
      itemsCount: 10
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
    wrapper.find("CartIconContainer").simulate("click");
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it("should not call toggleCartHidden when CartIcon is clicked, if itemsCount === 0", () => {
    const mockToggleCartHidden = jest.fn();
    const mockProps = {
      location: {
        pathname: "/"
      },
      toggleCartHidden: mockToggleCartHidden,
      itemsCount: 0
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
    wrapper.find("CartIconContainer").simulate("click");
    expect(mockToggleCartHidden).not.toHaveBeenCalled();
  });

  it('should not call toggleCartHidden when CartIcon is clicked, if pathname === "/checkout"', () => {
    const mockToggleCartHidden = jest.fn();
    const mockProps = {
      location: {
        pathname: "/checkout"
      },
      toggleCartHidden: mockToggleCartHidden,
      itemsCount: 10
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
    wrapper.find("CartIconContainer").simulate("click");
    expect(mockToggleCartHidden).not.toHaveBeenCalled();
  });
});
