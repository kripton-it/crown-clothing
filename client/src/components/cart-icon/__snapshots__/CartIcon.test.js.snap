// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should render 1`] = `ShallowWrapper {}`;
